debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/FindInstances.java

```diff
@@ -0,0 +1,165 @@
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ */
+package com.ibm.j9ddr.vm29.tools.ddrinteractive.commands;
+
+import java.io.PrintStream;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.tools.ddrinteractive.CommandUtils;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.tools.ddrinteractive.Table;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ObjectPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ClassHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+
+import com.ibm.j9ddr.vm29.j9.DataType;
+import com.ibm.j9ddr.vm29.j9.walkers.ClassSegmentIterator;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+
+import com.ibm.j9ddr.vm29.j9.gc.GCHeapRegionDescriptor;
+import com.ibm.j9ddr.vm29.j9.gc.GCHeapRegionIterator;
+import com.ibm.j9ddr.vm29.j9.gc.GCObjectHeapIterator;
+
+public class FindInstances extends Command
+{
+	private static final String nl = System.getProperty("line.separator");
+
+	private String className;
+	private J9ClassPointer classPointer;
+	private Table data;
+	private long corruptCount = 0;
+	private long objectsFound = 0;
+
+	public FindInstances() {
+		addCommand("findinstances", "<classname>", "find all instances of the specific class and its subclasses");
+	}
+
+	private void printUsage (PrintStream out) {
+		out.println("findinstances <name> - find all instances of the specific class and its subclasses");
+	}
+
+	private boolean parseArgs(PrintStream out, String[] args) throws DDRInteractiveCommandException {
+		if (args.length != 1) {
+			out.append("Invalid number of arguments" + nl);
+			return false;
+		}
+		String firstArg = args[0];
+		if (firstArg.equals("help")) {
+			printUsage(out);
+			return false;
+		}
+		className = firstArg;
+		return true;
+	}
+
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException
+	{
+		if (args != null) {
+			if (!parseArgs(out, args)) {
+				return;
+			}
+		}
+
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			ClassSegmentIterator iterator = new ClassSegmentIterator(vm.classMemorySegments());
+
+			out.println(
+				String.format(
+					"Searching for class named '%1$s' in VM=%2$s",
```

Comment:
Use `%n` in the format string:
```java
			out.format(
					"Searching for class named '%s' in VM=%s%n",
					className,
					Long.toHexString(vm.getAddress()));
```
and avoid unnecessary uses of `1$`, `2$`, etc.